packages/reputation-miner/ReputationMinerClient.js

```diff
@@ -0,0 +1,143 @@
+    super(_minerAddress, _realProviderPort);
+
+    // this.realProvider = new ethers.providers.InfuraProvider("kovan");
+
+    this._app = express();
+    this._app.get("/:colonyAddress/:skillId/:userAddress", async (req, res) => {
+      const key = await ReputationMiner.getKey(req.params.colonyAddress, req.params.skillId, req.params.userAddress);
+      if (key) {
+        const proof = await this.getReputationProofObject(key);
+        delete proof.nNodes;
+        proof.reputationAmount = ethers.utils.bigNumberify(`0x${proof.value.slice(2, 66)}`).toString();
+
+        res.status(200).send(proof);
+      } else {
+        res.status(400).send({ message: "Requested reputation does not exist or invalid request" });
+      }
+    });
+
+    this.server = this._app.listen(3000, () => {
+      console.log("⭐️ Reputation oracle running on port ", this.server.address().port);
+    });
+  }
+
+  /**
+   * Initialises the mining client so that it knows where to find the `ColonyNetwork` contract
+   * @param  {string}  colonyNetworkAddress The address of the current `ColonyNetwork` contract
+   * @return {Promise}
+   */
+  async initialise(colonyNetworkAddress) {
+    await super.initialise(colonyNetworkAddress);
+
+    try {
+      this.reputations = jsonfile.readFileSync(file);
+      console.log("💾 Restored from JSON file");
+    } catch (err) {
+      this.reputations = {};
+      console.log("No existing reputations found - starting from scratch");
+    }
+    this.nReputations = Object.keys(this.reputations).length;
+
+    if (this.nReputations === 0 && argv.seed) {
+      // Temporary data if --seed is set and there's nothing to restore from.
+      const ADDRESS1 = "0x309e642dbf573119ca75153b25f5b8462ff1b90b";
+      const ADDRESS2 = "0xbc13dbc1a954b3443d6f75297a232faa513774b3";
+      const ADDRESS3 = "0x2b183746bd1403cdec8e4fe45139339da20bcf3d";
+      const ADDRESS4 = "0xcd0751d4181acda4f8edb2f3b33b915f91abeef0";
+      const ADDRESS0 = "0x0000000000000000000000000000000000000000";
+      await this.insert(ADDRESS1, 1, ADDRESS2, "999999999");
+      await this.insert(ADDRESS1, 1, ADDRESS0, "999999999");
+      await this.insert(ADDRESS1, 2, ADDRESS2, "888888888888888");
+      await this.insert(ADDRESS1, 2, ADDRESS0, "888888888888888");
+      await this.insert(ADDRESS3, 1, ADDRESS2, "100000000");
+      await this.insert(ADDRESS3, 1, ADDRESS4, "100000000");
+      await this.insert(ADDRESS3, 1, ADDRESS0, "200000000");
+      console.log("💾 Writing initialised state with dummy data to JSON file");
+
+      jsonfile.writeFileSync(file, this.reputations);
+    } else {
+      for (let i = 0; i < Object.keys(this.reputations).length; i += 1) {
+        const key = Object.keys(this.reputations)[i];
+        await this.reputationTree.insert(key, this.reputations[key], { gasLimit: 4000000 }); // eslint-disable-line no-await-in-loop
+      }
+    }
+
+    console.log("🏁 Initialised");
+    const that = this;
```

Comment:
A cool feature of the new ES6 fat arrow is that it automatically binds the function to the outer context of it. So no need for `that` or `self` anymore. Hurray!